src/changelog.ts

```diff
@@ -118,44 +200,87 @@ export async function generateChangelog(
   if (bump.foundVersion === null) {
     return "";
   }
-  const config: Map<SemVerType, IChangelogCategory> =
-    getChangelogConfiguration();
-
+  const config = await getChangelogConfiguration();
   const { owner, repo } = context.repo;
 
   for (const commit of bump.messages) {
-    let msg = `${commit.description
-      .charAt(0)
-      .toUpperCase()}${commit.description.slice(1)}`;
+    const bumpLabel = `bump:${SemVerType[commit.bump].toLowerCase()}`;
+    const exclusionPatterns: IExcludeConfiguration = {
+      labels: [],
+      authors: [],
+    };
 
-    msg += await getPullRequestSuffix(commit);
-    msg += getIssueReferenceSuffix(commit);
+    // We will reuse the labels and author associated with a Pull Request
+    // (with the exception of `bump:<version`) for all commits associated
+    // with the PR.
 
     if (commit.hexsha) {
-      const sha_link = `[${commit.hexsha.slice(
-        0,
-        6
-      )}](https://github.com/${owner}/${repo}/commit/${commit.hexsha})`;
-      msg += ` [${sha_link}]`;
+      const pullRequests = await getAssociatedPullRequests(commit.hexsha);
+
+      if (pullRequests.length > 0) {
+        const pullRequest = pullRequests[0];
+
+        // Check the labels in the Pull Request associated with this commit
+        for (const label of pullRequest.labels) {
+          // Ignore bump-labels on Pull Requests
+          if (label.name.startsWith("bump:")) {
+            continue;
+          }
+
+          exclusionPatterns.labels?.push(label.name);
+        }
+        // Check for the author of the Pull Request
+        if (pullRequest.user) {
+          exclusionPatterns.authors?.push(pullRequest.user.login);
+        }
+      }
+    }
+
+    // Check the individual commit Bump Level
+    exclusionPatterns.labels?.push(bumpLabel);
+
+    if (hasConfigurationElement(exclusionPatterns, config.changelog.exclude)) {
+      continue;
     }
 
-    config.get(commit.bump)?.changes.push(msg);
+    for (const category of config.changelog.categories) {
```

Comment:
This seems unnecessary to do on every commit, as the config data never changes.

Try getting rid of this for-loop, e.g. consider a bulking up the data from `config.changelog.categories` once and doing a single call to `hasConfigurationElement()`.

Reply:
As discussed - will make a separate task to improve error handling (i.e. find duplications, ..), which will also rely on remapping data entries.